Grammar/python.gram

```diff
@@ -876,23 +886,59 @@ invalid_import_from_targets:
 invalid_with_stmt:
     | [ASYNC] 'with' ','.(expression ['as' star_target])+ &&':'
     | [ASYNC] 'with' '(' ','.(expressions ['as' star_target])+ ','? ')' &&':'
-
-invalid_except_block:
+invalid_with_stmt_indent:
+    | [ASYNC] a='with' ','.(expression ['as' star_target])+ ':' NEWLINE !INDENT {
+        RAISE_INDENTATION_ERROR("expected an indented block after with statement in line %d", a->lineno) }
```

Comment:
Sorry to make you change all these but I'd suggest changing the wording to

`... statement on line %d` instead of "in"